source/module_hamilt_general/module_vdw/dftd3_xc_name.h

```diff
@@ -0,0 +1,605 @@
+        {"XC_HYB_MGGA_XC_MPWKCIS1K", "mpwkcis1k"},
+        {"XC_HYB_MGGA_XC_PBE1KCIS", "pbe1kcis"},
+        {"XC_HYB_MGGA_XC_TPSS1KCIS", "tpss1kcis"},
+        {"XC_HYB_GGA_XC_B5050LYP", "b5050lyp"},
+        {"XC_LDA_XC_GDSMFB", "gdsmfb"},
+        {"XC_GGA_XC_KT3", "kt3"},
+        {"XC_HYB_LDA_XC_BN05", "bn05"},
+        {"XC_HYB_GGA_XC_LB07", "lb07"},
+        {"XC_HYB_MGGA_XC_B98", "b98"},
+        {"XC_LDA_XC_TIH", "tih"},
+        {"XC_HYB_GGA_XC_APBE0", "apbe0"},
+        {"XC_HYB_GGA_XC_HAPBE", "hapbe"},
+        {"XC_HYB_GGA_XC_RCAM_B3LYP", "rcam_b3lyp"},
+        {"XC_HYB_GGA_XC_WC04", "wc04"},
+        {"XC_HYB_GGA_XC_WP04", "wp04"},
+        {"XC_HYB_GGA_XC_CAMH_B3LYP", "camh_b3lyp"},
+        {"XC_HYB_GGA_XC_WHPBE0", "whpbe0"},
+        {"XC_HYB_GGA_XC_LC_BLYP_EA", "lc_blyp_ea"},
+        {"XC_HYB_GGA_XC_LC_BOP", "lc_bop"},
+        {"XC_HYB_GGA_XC_LC_PBEOP", "lc_pbeop"},
+        {"XC_HYB_GGA_XC_LC_BLYPR", "lc_blypr"},
+        {"XC_HYB_GGA_XC_MCAM_B3LYP", "mcam_b3lyp"},
+        {"XC_MGGA_XC_VCML_RVV10", "vcml_rvv10"},
+        {"XC_HYB_MGGA_XC_GAS22", "gas22"},
+        {"XC_HYB_MGGA_XC_R2SCANH", "r2scanh"},
+        {"XC_HYB_MGGA_XC_R2SCAN0", "r2scan0"},
+        {"XC_HYB_MGGA_XC_R2SCAN50", "r2scan50"},
+        {"XC_HYB_GGA_XC_CAM_PBEH", "cam_pbeh"},
+        {"XC_HYB_GGA_XC_CAMY_PBEH", "camy_pbeh"},
+        {"XC_HYB_MGGA_XC_EDMGGAH", "edmggah"},
+        {"XC_HYB_MGGA_XC_LC_TMLYP", "lc_tmlyp"},
+    };
+    const std::map<std::string, std::string> xcname_libxc_xplusc_ = {
+        {"XC_GGA_X_GAM+XC_GGA_C_GAM", "gam"},
+        {"XC_GGA_X_HCTH_A+XC_GGA_C_HCTH_A", "hcth_a"},
+        {"XC_HYB_MGGA_X_DLDF+XC_MGGA_C_DLDF", "dldf"},
+        {"XC_GGA_X_Q2D+XC_GGA_C_Q2D", "q2d"},
+        {"XC_GGA_X_PBE_MOL+XC_GGA_C_PBE_MOL", "pbe_mol"},
+        {"XC_GGA_X_PBEINT+XC_GGA_C_PBEINT", "pbeint"},
+        {"XC_HYB_GGA_X_N12_SX+XC_GGA_C_N12_SX", "n12_sx"},
+        {"XC_GGA_X_N12+XC_GGA_C_N12", "n12"},
+        {"XC_GGA_X_PBE+XC_GGA_C_PBE", "pbe"},
+        {"XC_GGA_X_B88+XC_MGGA_C_B88", "b88"},
+        {"XC_GGA_X_PW91+XC_GGA_C_PW91", "pw91"},
+        {"XC_GGA_X_PBE_SOL+XC_GGA_C_PBE_SOL", "pbe_sol"},
+        {"XC_GGA_X_AM05+XC_GGA_C_AM05", "am05"},
+        {"XC_GGA_X_XPBE+XC_GGA_C_XPBE", "xpbe"},
+        {"XC_GGA_X_RGE2+XC_GGA_C_RGE2", "rge2"},
+        {"XC_GGA_X_SOGGA11+XC_GGA_C_SOGGA11", "sogga11"},
+        {"XC_GGA_X_APBE+XC_GGA_C_APBE", "apbe"},
+        {"XC_MGGA_X_TPSS+XC_MGGA_C_TPSS", "tpss"},
+        {"XC_MGGA_X_M06_L+XC_MGGA_C_M06_L", "m06_l"},
+        {"XC_HYB_MGGA_X_TAU_HCTH+XC_GGA_C_TAU_HCTH", "tau_hcth"},
+        {"XC_MGGA_X_REVTPSS+XC_MGGA_C_REVTPSS", "revtpss"},
+        {"XC_MGGA_X_PKZB+XC_MGGA_C_PKZB", "pkzb"},
+        {"XC_MGGA_X_M11_L+XC_MGGA_C_M11_L", "m11_l"},
+        {"XC_MGGA_X_MN12_L+XC_MGGA_C_MN12_L", "mn12_l"},
+        {"XC_HYB_MGGA_X_MN12_SX+XC_MGGA_C_MN12_SX", "mn12_sx"},
+        {"XC_MGGA_X_MN15_L+XC_MGGA_C_MN15_L", "mn15_l"},
+        {"XC_MGGA_X_SCAN+XC_MGGA_C_SCAN", "scan"},
```

Comment:
For ABACUS user who use LibXC functionals, there are two ways for setting `dft_functional`, take SCAN as an example:
1. XC_MGGA_X_SCAN+XC_MGGA_C_SCAN
2. MGGA_X_SCAN+MGGA_C_SCAN. which leave the XC_ head behind. 

But in this code, the second way seems to be not supported